components/FilterBlock/FilterBlock.tsx

```diff
@@ -0,0 +1,360 @@
+import { Checkbox, Collapse } from "antd";
+import "antd/dist/reset.css";
+
+import { Fragment, useState, useEffect } from "react";
+
+import { CheckBoxProps, SelectedShift } from "../../types";
+
+type FilterBlockProps = {
+  layer1: string[]; // contains the titles for the collapse's of the 1st layer
+  layer2?: string[]; // contains the titles for the collapse's of the 2nd layer
+  checkBoxes: CheckBoxProps[][]; // contains the checkboxes information in a universal format
+  exception?: number; // indicates the index of an element from layer1 where the layer2 should be ignored, for example "5th year"
+  checked: any; // assumes different types when called from ScheduleFilters.tsx and EventFilters.tsx
```

Comment:
You could create a union of types, then.

Reply:
Done.